### DISCUSSION TITLE
[manifest] Output error message when json parsing fails

### DISCUSSION THREAD

**Discussion by timmeh87** [microsoft/vcpkg#19196] [Ideas]
see comments thread for the actual bug

**Describe the bug**
I noticed that I can craft a manifest file that is clearly invalid JSON. Usually vcpkg will output a red message with the exact json error, but if I put certain $comments into the JSON file, vcpkg fails with no output at all.

**Environment**
- OS: win10 19042
- Compiler: vs2019 community

**To Reproduce**
This manifest file contains two comments, and causes the behavior for me.
```
{
    "name":"name",
    "$comment:":"hi",
    "version":"1.0.0",
    "$comment:":"hi",
    "builtin-baseline":"eefee7408133f3a0fef711ef9c6a3677b7e06fd7",
    "dependencies":[
        {"name":"zlib"}
    ],
    "overrides": [
        {"name":"zlib", "version-string":"1.2.11", "port-version":10},
    ]
}
```
vcpkg crashes without any output when I give it this file using this exact command:
`E:/new_libs/scripts/vcpkg/vcpkg.exe install --feature-flags="manifests,versions" --triplet="x64-windows" --downloads-root="E:/new_libs/scripts/vcpkg-downloads-init" --overlay-ports="overlay" --x-install-root="e:/new_libsrelease/output//vs-2019_x64/vcpkg-libs/vcpkg-libs-5-12-21/"`

 If I  remove either comment from that and use a single comment,  I get this error:

```diff
- Failed to parse manifest at E:\new_libs\libs\vcpkg-libs\windows-x64\vcpkg.json:
- E:\new_libs\libs\vcpkg-libs\windows-x64\vcpkg.json:10:78: error: Trailing comma in array
-   on expression:               {"name":"zlib", "version-string":"1.2.11", "port-version":10},
-                                                                                             ^
- Note: Updating vcpkg by rerunning bootstrap-vcpkg may resolve this failure.
- which has nothing to do with the comment
```

**Expected behavior**
An error every time regardless of the comments

**note**
The json has CRLF line endings and no tabs (just spaces)

**Comment by JackBoosY** [comment]
Should be:
```json
{
    "name": "name", 
    "$comment:": "hi", 
    "version": "1.0.0", 
    "builtin-baseline": "eefee7408133f3a0fef711ef9c6a3677b7e06fd7", 
    "dependencies": [
        {
            "name": "zlib"
        }
    ], 
    "overrides": [
        {
            "name": "zlib", 
            "version-string": "1.2.11", 
            "port-version": 10
        }
    ]
}
```
?

**Comment by timmeh87** [comment]
Jack: Yes, that would be the correct manifest file that causes vcpkg to execute correctly. 

The issue here is that if the manifest has invalid JSON, and it contains "certain comments" (in my case, it seemed to only happen if there was more than 1 comment) then vcpkg outputs *nothing*

If it were to output the red error message in my original post, It would be obvious that the comma needed to be removed from the manifest file to correct it. But I spent like an hour looking for a log file or *something* to explain why vcpkg was producing no output at all. I then found out that it normally outputs a JSON error message that is very helpful. But if there are "certain comments" that are themselves valid JSON,  then the JSON error does not get printed. I believe this is a bug.

**Comment by dg0yt** [comment]
> it contains "certain comments"

I think the problem isn't *certain comments* (error in handling comments) but *duplicate keys* (invalid JSON).

**Comment by timmeh87** [comment]
If duplicate keys makes an invalid json file (doh! yeah that makes sense) then the fix is pretty simple, it should output a red error message similar to "JSON error: duplicate key on line xxxx"

**Comment by timmeh87** [comment]
I can reproduce by duplicating any of the JSON keys

**Comment by dg0yt** [comment]
OTOH it is perhaps more important to improve vcpkg behaviour for valid data then to improve behaviour for invalid data. I suppose there are other tools for syntactical validation. (Typing `jsonlint + <TAB>` on my command prompt suggests `jsonlint-php`.)
